learn-typescript-linting.md

```diff
@@ -0,0 +1,407 @@
+# How to Set Up ESLint for TypeScript
+
+Written by: Ethan Arrowood
+
+This guide will show you how to set up [ESLint](https://eslint.org/) with a [TypeScript](https://www.typescriptlang.org/) project. The guide is broken up into five sections. The first two are about setting up ESLint and configuring it to work with TypeScript. The third and forth sections are for integrating popular styling formatters [Standard](https://standardjs.com/) and [Prettier](https://prettier.io/). The last section contains additional context and a list of resources for those interested in learning more. Make use of the table of contents and the `[toc]` shortcuts to better navigate this article.
+
+### Table of Contents
+- [1 Getting Started](#1-getting-started)
+- [2 Adding ESLint](#2-adding-eslint)
+  - [2.1 Initializing .eslintrc](#21-initializing-eslintrc)
+  - [2.2 Specifying environments](#22-specifying-environments)
+  - [2.3 Specifying ecmaVersion](#23-specifying-ecmaVersion)
+  - [2.4 Creating ESLint npm script](#24-Creating-ESLint-npm-script)
+  - [2.5 Executing ESLint](#25-executing-eslint)
+  - [2.6 Fixing an ESLint warning](#26-fixing-an-eslint-warning)
+  - [2.7 Configuring ESLint](#27-configuring-eslint)
+  - [2.8 Additional ESLint rule configuration](#28-additional-eslint-rule-configuration)
+  - [2.9 Fixing unused variable definition error from type import](#29-fixing-unused-variable-definition-error-from-type-import)
+- [3 Adding Standard Style Formatter](#3-adding-standard-style-formatter)
+  - [3.1 Installing Standard](#31-installing-standard)
+  - [3.2 Evaluating new errors](#32-evaluating-new-errors)
+  - [3.3 Configuring Standard specific rules](#33-configuring-standard-specific-rules)
+- [4 Adding Prettier Style Formatter](#4-adding-prettier-style-formatter)
+  - [4.1 Installing Prettier](#41-installing-prettier)
+  - [4.2 Executing Prettier](#42-executing-prettier)
+  - [4.3 Configuring Prettier specific rules](#43-configuring-prettier-specific-rules)
+- [5 Additional Resources and Documentation](#5-additional-resources-and-documentation)
+
+## 1 Getting Started
+[`[toc]`](#table-of-contents)
+
+**Note:*** Developers without an existing TypeScript project should start here at section 1; developers with an existing project can skip ahead to section [2 Adding ESLint](#2-adding-eslint). This guide works best if you follow along with the GitHub repository.
+
+View the GitHub repository [`learn-typescript-linting`](https://github.com/MatterhornDev/learn-typescript-linting). Copy, paste and execute the following command to clone it to your machine:
+
+```bash
+git clone git@github.com:MatterhornDev/learn-typescript-linting.git
```

Comment:
This didn't work for me, I had to use:
`git clone https://github.com/MatterhornDev/learn-typescript-linting.git`
Then I moved into the folder then changed branches